apps/ca.c

```diff
@@ -1177,10 +1178,9 @@ int ca_main(int argc, char **argv)
         }
 
         /* we have a CRL number that need updating */
-        if (crlnumberfile != NULL)
-            if (!rand_ser
```

Comment:
@richsalz please double-check: the `if (!rand_ser` here appears wrong to me: While the conditions `serialfile != NULL` and `rand_ser != NULL` are mutually exlusive, the conditions  `crlnumberlfile != NULL` and `rand_ser != NULL` are seemingly unrelated. So I removed this extra check.